ui/desktop/app/helpers/time-remaining.js

```diff
@@ -0,0 +1,44 @@
+import Helper from '@ember/component/helper';
+import { inject as service } from '@ember/service';
+
+export default class extends Helper {
+  // =services
+  @service clockTick;
+  @service intl;
+
+  compute([expiration_time]) {
+    if (!expiration_time) return;
+
+    // Parse the expiration time
+    const expirationTime = new Date(expiration_time).getTime();
+
+    // Get the curent time
+    const currentTime = this.clockTick.now;
```

Comment:
I don't think we should take the current time but use the `created_time` we receive within the payload. I will try to explain better:

- Once a session is created we receive the session payload, wich includes `created_time` and `expiration_time`.
- The remainig time of a session is not: current time - expiration time. I think it is: creation time - expiration time.
- I do think the creation time does not need to be necessary the current time. For instance, a user can create a session, do not cancel it, navigate around the desktop client and later on came back to the session detail, then, if we are using current time the remaining time will be incorrect.

Let me know your thoughts on this one or if you want to discuss further

Reply:
I deleted the previous comment. I was totally wrong.

Because expiration time is a timestamp to get remaining time your implementation is totally correct. remaining time = current time - expiration time.